### ISSUE TITLE
How to evaluate (compare) two different topic models results produced from top2vec model?

### ISSUE THREAD

**Issue by PiyushKyushu** [ddangelov/Top2Vec#104]
First of all, thanks for the top2vec model. It produces very coherent topics.

My question is more about seeking suggestions than any issue.

I have two similar datasets (same theme) - A and B. I used the top2vec model and it produces topics (let's say 76 and 98 number of topics). 

I want to compare topics in the A and B datasets computationally (for example using clustering or any other similar methods). Can anyone suggest some ideas or methods for comparing the topics? 

Also, can anyone suggest how to produce figures such as 8 and 14 (of top2vec paper)?

Thank you.

**Comment by UmutAlihan** [comment]
+1

**Comment by DrozdikGleb** [comment]
I see three possible ways for comparing the topic models:
1) Using topic coherence metric (from [gensim](https://radimrehurek.com/gensim/models/coherencemodel.html) lib, for example)
2) Using clustering metrics for unlabeled dataset, such as [silhoutte coefficient](https://scikit-learn.org/stable/modules/clustering.html#silhouette-coefficient), Calinski-Harabasz Index or Davies-Bouldin Index.
3) Using visualization. [Embedding projector](https://projector.tensorflow.org/) or similar to LDAvis tool from [BERTopic](https://github.com/MaartenGr/BERTopic#visualize-topics)

**Comment by p-dre** [comment]
In the paper from top2vec Mutual Information is used. My implementation for the application on top2vec is unfortunately very slow and much too impractical. Maybe @ddangelov can give a hint how a good implimentation like in the paper is possible.

**Comment by ddangelov** [comment]
See #17 #19 #30

There will eventually be an implementation of topic information gain however it is currently not a priority. It is slow, but it can be parallelized which I haven't done yet but that would be one way to speed it up.

**Comment by doodyology** [comment]
@p-dre Would you mind sharing your implementation of your PWI script, even if it is slow? Thanks!
